.github/workflows/poetry-test.yml

```diff
@@ -9,19 +9,30 @@ on:
 jobs:
   build-and-test-ros2:
     if: github.event.pull_request.draft == false
+
+    runs-on:
+      - self-hosted
+      - ${{ matrix.ros_distro }}
+
+    continue-on-error: true
+
     strategy:
       matrix:
-        include:
-          - ros_distro: jazzy
-            ubuntu_version: "24.04"
-          - ros_distro: humble
-            ubuntu_version: "22.04"
-    runs-on: ubuntu-${{ matrix.ubuntu_version }}
+        ros_distro:
+          - jazzy
+          - humble
 
     container:
       image: osrf/ros:${{ matrix.ros_distro }}-desktop-full
 
     steps:
+      - name: Clean up workspace
+        run: |
+          ls -la $GITHUB_WORKSPACE
+          rm -rf $GITHUB_WORKSPACE/* || true
+          rm -rf $GITHUB_WORKSPACE/.??* || true
+          ls -la $GITHUB_WORKSPACE
+
```

Comment:
Is this step needed?

Reply:
I think so, as it cleans the repo with build, install & log directories (check workflows logs for the reference). `ls -la $GITHUB_WORKSPACE` can be removed as it only lists files before and after the clean-up however, it makes it easier to debug and validate if `rm -rf` command actually worked.